posts/lte-examples.md

```diff
@@ -0,0 +1,692 @@
+These examples can be considered as centralized "sanity checks" that the definitions we wrote using Lean actually behave as expected.
+
+We tried to write the files in this folder in a way which should be (approximately) readable by mathematicians who have minimal experience with Lean.
+The goal is to make it easy for non-experts to look through the examples folder, then look through the concise final statement in `challenge.lean`, and be confident to a reasonable extent that the challenge was accomplished.
+
+This blog post gives a detailed overview of this folder and its contents, and how it relates to the definitions used in the main statement of the [liquid tensor experiment (LTE)](https://github.com/leanprover-community/lean-liquid).
+It is meant to be read *alongside* the actual files from the [examples folder](https://github.com/leanprover-community/lean-liquid/tree/13777e1f84a324030eed48e636550aee90f8656f/src/examples).
+
+The links below involving LTE all point to files in [this commit](https://github.com/leanprover-community/lean-liquid/commit/13777e1f84a324030eed48e636550aee90f8656f) of the LTE repository, which is the most recent one as of writing this post.
+
+<!-- TEASER_END -->
+
+# Definitions and examples in Lean
+
+This post contains several Lean code blocks. 
+Most of them have only `example`s, but a few also contain `def`enitions.
+Let's briefly review how to read such a piece of code.
+Lean experts can safely skip this section.
+
+## The anatomy of definition
+
+Let's take a look at the following code block:
+```lean
+def translate_by_pos (a : ℕ) (ha : 0 < a) : 
+  ℕ → ℕ := 
+λ t, a + t
+```
+In this case, we are making a *definition* which can be later accessed with the name `translate_by_pos`.
+The code to the left of the `:` on the first line can be thought of as "hypotheses" for the definition, which in this case consists of a natural number `a` and a hypothesis `ha` that `a` is positive.
+
+The code after the `:` at the end of the first line and before the `:=` on the second line tells Lean what to expect from the actual definition.
+In this case, we are defining a function from `ℕ` to itself, and the collection of such functions is denoted with an arrow as `ℕ → ℕ`.
+
+The text following `:=` is the actual definition which in this case is `λ t, a + t`.
+This is an (unnamed) function declared with notation from lambda calculus.
+A mathematician might write $t \mapsto a + t$ instead.
+
+## Examples vs definitions
+
+An example is just like a definition, except that it will not add any declaration to the current environment, so it does not require that a name is provided.
+For instance, if we did not need to refer to `translate_by_pos` in the future, we could have just written
+```lean
+example (a : ℕ) (ha : 0 < a) : 
+  ℕ → ℕ := 
+λ t, a + t
+```
+This can still useful because it tells us that `λ t, a + t` is a valid element of `ℕ → ℕ`, given the hypotheses on the first line. 
```

Comment:
```suggestion
This can still be useful because it tells us that `λ t, a + t` is a valid element of `ℕ → ℕ`, given the hypotheses on the first line. 
```